modules/stereo/src/descriptor.hpp

```diff
@@ -0,0 +1,435 @@
+
+#include "precomp.hpp"
+#include <stdint.h>
+#ifndef _OPENCV_DESCRIPTOR_HPP_
+#define _OPENCV_DESCRIPTOR_HPP_
+#ifdef __cplusplus
+#define CV_DENSE_CENSUS 0
+#define CV_SPARSE_CENSUS 1
+#define CV_MODIFIED_CENTER_SIMETRIC_CENSUS 0
+#define CV_CLASSIC_CENTER_SYMETRIC_CENSUS 1
+#define CV_STANDARD_MCT 0
+#define CV_MCT_MEAN_VARIATION 1
+#define CV_SSE_CENSUS 1
+
+namespace cv
+{
+	namespace stereo
+	{
+		//!class implemented to run the census descriptor in parralel
+		class parallelCensus :public ParallelLoopBody
+		{
+		private:
+			uint8_t *image1, *image2;
+			int *dst1, *dst2;
+			int n2, width, height, type;
+		public:
+			parallelCensus(uint8_t * img1, uint8_t * img2, int w, int h, int k2, int * distance1, int * distance2,const int t) :
+				image1(img1), image2(img2), dst1(distance1), dst2(distance2), n2(k2), width(w), height(h), type(t){}
+			virtual void operator()(const cv::Range &r) const {
+				for (int i = r.start; i <= r.end ; i++)
+				{
+					int rWidth = i * width;
+					for (int j = n2; j <= width - n2; j++)
+					{
+						//imitialize the costs for the 2 census descriptors
+						int c = 0;
+						int c2 = 0;
+						//verify if user wants a dense census
+						if (type == CV_DENSE_CENSUS)
+						{
+							for (int ii = i - n2; ii <= i + n2; ii++)
+							{
+								int rrWidth = ii * width;
+								for (int jj = j - n2; jj <= j + n2; jj++)
+								{
+									if (ii != i || jj != j)
+									{
+										//compare a pixel with the center from the kernel
+										if (image1[rrWidth + jj] > image1[rWidth + j])
+										{
+											c = c + 1;
+										}
+										c = c * 2;
+									}
+									if (ii != i || jj != j)
+									{
+										//compare pixel with center for image 2
+										if (image2[rrWidth + jj] > image2[rWidth + j])
+										{
+											c2 = c2 + 1;
+										}
+										c2 = c2 * 2;
+									}
+								}
+							}
+						}//verifi if user wants a sparser census
+						else if (type == CV_SPARSE_CENSUS)
+						{
+							for (int ii = i - n2; ii <= i + n2; ii += 2)
+							{
+								int rrWidth = ii * width;
+								for (int jj = j - n2; jj <= j + n2; jj += 2)
+								{
+									if (ii != i || jj != j)
+									{
+										//compare center pixel with pixel at distance 2x
+										if (image1[(rrWidth + jj)] > image1[(rWidth + j)])
+										{
+											c = c + 1;
+										}
+										c = c * 2;
+									}
+									if (ii != i || jj != j)
+									{
+										//same for image 2
+										if (image2[(rrWidth + jj)] > image2[(rWidth + j)])
+										{
+											c2 = c2 + 1;
+										}
+										c2 = c2 * 2;
+									}
+								}
+							}
+						}
+						dst1[(rWidth + j)] = c;
+						dst2[(rWidth + j)] = c2;
+					}
+				}
+			}
+		};
+		//!class that implemented the census descriptor on single images
+		class singleImageCensus : public ParallelLoopBody
```

Comment:
I'm a bit confused here. As far as I can see, parallelCensus is the exact same thing as performing singleImageCesus on 2 images consequently. If this is so, why do you need parallelCensus at all?